client/src/main/scala/org/http4s/client/digest/SimpleInMemoryStoreNonceCounter.scala

```diff
@@ -1,18 +1,23 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
 package org.http4s.client.digest
 
 import java.util.concurrent.TimeUnit
 
 import cats.effect.concurrent.Ref
 import cats.effect.{Concurrent, Timer}
 import cats.implicits._
-import org.http4s.client.digest.NonceCounter.MaxNonce
 
 import scala.concurrent.duration.FiniteDuration
 
-class SimpleInMemoryStoreNonceCounter[F[_]](nonceTtl: FiniteDuration, cnonceF: () => String)(
-    implicit
-    T: Timer[F],
-    C: Concurrent[F])
+class SimpleInMemoryStoreNonceCounter[F[_]](
+    nonceTtl: FiniteDuration,
+    cnonceF: () => String,
```

Comment:
I think this should be an `F[String]` to reflect its impurity.  Sequencing that into `val current` is easy.  Getting a second one into `next`, without calculating it in advance, is a bit of a brain teaser.